.github/workflows/ci-code-analysis.yml

```diff
@@ -0,0 +1,60 @@
+on:
+  workflow_call:
+    inputs:
+      checkout-repo:
+        required: false
+        type: string
+      checkout-ref:
+        required: false
+        type: string
+
+permissions:
+  contents: read
+  packages: read
+
+jobs:
+  spotless:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        repository: ${{ inputs.checkout-repo }}
+        ref: ${{ inputs.checkout-ref }}
+    - name: Run spotless
+      run: mvn -B -U spotless:check
+
+  spotbugs:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        repository: ${{ inputs.checkout-repo }}
+        ref: ${{ inputs.checkout-ref }}
+        submodules: true
+        fetch-depth: 0
+    - uses: skjolber/maven-cache-github-action@v1
+      with:
+        step: restore
+    - uses: actions/setup-java@v3
+      with:
+        java-version: '17'
+        distribution: 'temurin'
+    - name: maven-settings
+      uses: s4u/maven-settings-action@v2
+      with:
+        servers: '[{"id": "github", "username": "dummy", "password": "${env.GITHUB_TOKEN_REF}"}]'
+        githubServer: false
+    - name: Run spotbugs
+      run: mvn -B -U -Dheadless=true compile spotbugs:check
```

Comment:
I don't think `-Dheadless` works with the new build setup. Maybe `-Dquarkus.quinoa=false` would be an equivalent, but I'm not sure if that actually prevents it from building the assets or just turns off the devmode webserver.

https://docs.quarkiverse.io/quarkus-quinoa/dev/config-reference.html